# Fix for "[filter avfilter.gblur filter6] Cannot set src colorspace 2" error with ffmpeg6; no upstream commit exists.
# mlt passes "colorspace" or "range" options to ffmpeg which do not exist in ffmpeg6
diff --git a/src/modules/avformat/filter_avfilter.c b/src/modules/avformat/filter_avfilter.c
--- a/src/modules/avformat/filter_avfilter.c
+++ b/src/modules/avformat/filter_avfilter.c
@@ -504,15 +504,19 @@ static void init_image_filtergraph(mlt_filter filter,
         mlt_log_error(filter, "Cannot set src frame_rate %d/%d\n", framerate.num, framerate.den);
         goto fail;
     }
-    ret = av_opt_set_int(pdata->avbuffsrc_ctx, "colorspace", avcolorspace, AV_OPT_SEARCH_CHILDREN);
-    if (ret < 0) {
-        mlt_log_error(filter, "Cannot set src colorspace %d\n", avcolorspace);
-        goto fail;
+    if (av_opt_find(pdata->avbuffsrc_ctx, "colorspace", NULL, 0, AV_OPT_SEARCH_CHILDREN)) {
+        ret = av_opt_set_int(pdata->avbuffsrc_ctx, "colorspace", avcolorspace, AV_OPT_SEARCH_CHILDREN);
+        if (ret < 0) {
+            mlt_log_error(filter, "Cannot set src colorspace %d\n", avcolorspace);
+            goto fail;
+        }
     }
-    ret = av_opt_set_int(pdata->avbuffsrc_ctx, "range", color_range, AV_OPT_SEARCH_CHILDREN);
-    if (ret < 0) {
-        mlt_log_error(filter, "Cannot set src range %d\n", color_range);
-        goto fail;
+    if (av_opt_find(pdata->avbuffsrc_ctx, "range", NULL, 0, AV_OPT_SEARCH_CHILDREN)) {
+        ret = av_opt_set_int(pdata->avbuffsrc_ctx, "range", color_range, AV_OPT_SEARCH_CHILDREN);
+        if (ret < 0) {
+            mlt_log_error(filter, "Cannot set src range %d\n", color_range);
+            goto fail;
+        }
     }
     ret = avfilter_init_str(pdata->avbuffsrc_ctx, NULL);
     if (ret < 0) {
diff --git a/src/modules/avformat/link_avfilter.c b/src/modules/avformat/link_avfilter.c
--- a/src/modules/avformat/link_avfilter.c
+++ b/src/modules/avformat/link_avfilter.c
@@ -520,15 +520,19 @@ static void init_image_filtergraph(mlt_link self,
         mlt_log_error(self, "Cannot set src frame_rate %d/%d\n", framerate.num, framerate.den);
         goto fail;
     }
-    ret = av_opt_set_int(pdata->avbuffsrc_ctx, "colorspace", avcolorspace, AV_OPT_SEARCH_CHILDREN);
-    if (ret < 0) {
-        mlt_log_error(self, "Cannot set src colorspace %d\n", avcolorspace);
-        goto fail;
+    if (av_opt_find(pdata->avbuffsrc_ctx, "colorspace", NULL, 0, AV_OPT_SEARCH_CHILDREN)) {
+        ret = av_opt_set_int(pdata->avbuffsrc_ctx, "colorspace", avcolorspace, AV_OPT_SEARCH_CHILDREN);
+        if (ret < 0) {
+            mlt_log_error(self, "Cannot set src colorspace %d\n", avcolorspace);
+            goto fail;
+        }
     }
-    ret = av_opt_set_int(pdata->avbuffsrc_ctx, "range", color_range, AV_OPT_SEARCH_CHILDREN);
-    if (ret < 0) {
-        mlt_log_error(self, "Cannot set src range %d\n", color_range);
-        goto fail;
+    if (av_opt_find(pdata->avbuffsrc_ctx, "range", NULL, 0, AV_OPT_SEARCH_CHILDREN)) {
+        ret = av_opt_set_int(pdata->avbuffsrc_ctx, "range", color_range, AV_OPT_SEARCH_CHILDREN);
+        if (ret < 0) {
+            mlt_log_error(self, "Cannot set src range %d\n", color_range);
+            goto fail;
+        }
     }
     ret = avfilter_init_str(pdata->avbuffsrc_ctx, NULL);
     if (ret < 0) {
